File: String/2942.Find_Words_Containing_Character/Python/2942.Find_Words_Containing_Character.py
class Solution(object):
    def findWordsContaining(self, words, x):
        """
        :type words: List[str]
        :type x: str
        :rtype: List[int]
        """
        #========================================#
        # One-pass record based traversal method #
        #========================================#

        ############
        #Initialize
        ##### Length of words array #####
        len_words = len(words)

        ##### Result array #####
        res_arry = []


        ######################################
        #One-pass record based loop traversal
        for words_idx in range(len_words):

            ##### Check if the current indexed-char existed or not #####
            if (x not in words[words_idx]):
                continue

            res_arry.append(words_idx) #Keep updating/recording

        return res_arry

File: String/2942.Find_Words_Containing_Character/C++/2942.Find_Words_Containing_Character.cpp
class Solution {
public:
    vector<int> findWordsContaining(vector<string> & words, char x) 
    {
        //========================================//
        // Input type:                            //
        //   - vector<string> ref. (words vector) //
        //   - char (x char)                      //
        // Return type:                           //
        //   - vector<int> (result vector)        //
        //========================================//

        /*Initialize*/
        ///// Length of words vector /////
        int len_words = words.size();

        ///// Result vector /////
        vector<int> res_vect;


        /*One-pass record based loop traversal*/
        for (int words_idx = 0; (words_idx < len_words); (words_idx++)) //Whole
        {
            ///// Check if the current indexed-char existed or not /////
            if ((words[words_idx]).find(x) == string::npos) { continue; }

            res_vect.emplace_back(words_idx); //Keep updating, recording

        } //Whole

        return res_vect;
    }
};
